app/controllers/application_controller.rb

```diff
@@ -35,6 +35,10 @@ def authenticate_active_admin_user!
     end
   end
 
+  def after_sign_in_path_for(resource)
+    session[:download_info] ? download_and_redirect_practice_resources_path : super
+  end
+
```

Comment:
This is a `devise` method overwrite that allows the browser to nav to the originally intended destination after required authentication. We could actually leverage this for all auth-protected endpoints such as non-public innovation pages if we wanted, currently a non-logged in user is simply redirected to the home page with an "unauthorized" banner message appearing under the nav bar.